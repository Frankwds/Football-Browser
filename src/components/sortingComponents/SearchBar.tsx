import { Input, InputGroup, InputLeftAddon } from "@chakra-ui/react";
import React from "react";
import "./../style.css";

type Props = {
  get: (childData: string) => void;
  color: string;
  type: string;
  id: string;
};

/**
 * SearchBar - Input fields to be used as search parameters
 *
 * @param get
 * @param color
 * @param type
 *
 * @returns {React.FC}
 *
 * The @param get is a hook passed from @link {FilterBox.tsx}
 *
 */

const SearchBar: React.FC<Props> = ({ get, color, type, id }) => {
  return (
    <InputGroup>
      <InputLeftAddon
        children={type}
        background={color}
        color={"white"}
        border={"none"}
      />
      <Input
        id = {id}
        onChange={(event) => get(event.currentTarget.value)}
        placeholder={"All"}
        background={"white"}
        focusBorderColor={color}
      />
    </InputGroup>
  );
};

export default SearchBar;
