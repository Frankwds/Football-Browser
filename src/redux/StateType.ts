import {SearchData} from "./searchFilter/searchFilterTypes"; 

//A type interface for the only implemented query, scalable with more reducers.
export interface StateType{
    searchQueries: SearchData,
}


