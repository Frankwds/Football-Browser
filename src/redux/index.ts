import { combineReducers } from "redux";
import {searchReducer} from "./searchFilter/filterSearchReducer";


// In this case only one reducer, but very scalable.
const allReducers = combineReducers({
  searchQueries: searchReducer,
});

export default allReducers;