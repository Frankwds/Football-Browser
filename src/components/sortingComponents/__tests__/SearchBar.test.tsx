import { ChakraProvider } from "@chakra-ui/react";
import { useState } from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { createStore } from "redux";
import allReducers from "../../../redux";
import SearchBar from "../SearchBar";

const acc = "rgb(0,128,128)";
let container: any;
const store = createStore(allReducers);

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Testing SearchBar", () => {
  it("should render without crashing", () => {
    act(() => {
      const div = document.createElement("div");
      ReactDOM.render(
        <Provider store={store}>
          <ChakraProvider>
            <SearchBar testid={"iid"} color={acc} type={"test: "} get={() => {}} />
          </ChakraProvider>
        </Provider>,
        container
      );
    });
  });

  it("should render with the correct fields", () => {
    act(() => {
      ReactDOM.render(
        <Provider store={store}>
          <ChakraProvider>
            <SearchBar testid={"iid"} color={acc} type={"test: "} get={() => {}} />
          </ChakraProvider>
        </Provider>,
        container
      );
      expect(container.querySelector("div div").textContent).toBe("test: ");
    });
  });

  it("snapshot should be same as previous", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ChakraProvider>
            <SearchBar testid={"iid"} color={acc} type={"test: "} get={() => {}} />
          </ChakraProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should update text hook outside of component", () => {
    act(() => {
      let parentText = "";
      function setText(text: string) {
        parentText = text;
      }
      const div = document.createElement("div");
      ReactDOM.render(
        <Provider store={store}>
          <ChakraProvider>
            <SearchBar testid={"iid"} color={acc} type={"test: "} get={setText} />
          </ChakraProvider>
        </Provider>,
        container
      );
      console.log(parentText);
    });
  });
});
