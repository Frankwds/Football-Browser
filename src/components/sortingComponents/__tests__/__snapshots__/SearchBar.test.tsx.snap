// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchBar snapshot should be same as previous 1`] = `
<div
  className="chakra-input__group css-4302v8"
>
  <div
    className="chakra-input__left-addon css-7xv77h"
  >
    test: 
  </div>
  <input
    className="chakra-input css-pyk3ok"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="All"
  />
</div>
`;
